packages/upswyng-web/src/components/BackButton.tsx

```diff
@@ -29,14 +30,33 @@ const CategoryBannerArrowBack = styled(ArrowBack)`
 ` as typeof ArrowBack;
 
 const customHistory = createBrowserHistory();
+const resourceRegex: RegExp = /(resource\/)/;
```

Comment:
Rather than having the functionality of the back button be based on the current URL, what do you think about specifically passing a path to the `<BackButton/>` component?

This prop could be something like `sourceUrl` and have a default value of "/", which accounts for directly navigating to the page where the back-button is used. Otherwise, it would be passed down from wherever its source was. So if this back button is used within the `<ResourceList/>`, its `sourceUrl` would be the home page. If the back button is used on an individual `<Resource/>` page, then its `sourceUrl` would either be the list of resources by category/subcategory or the search results page.

That way, instead of coupling this back button to a URL that could change, we'll just explicitly tell it where to go based on where the user came from.

Reply:
Sounds perfect. I felt really 'dirty' parsing the url like that anyway 😂